pkg/controller/controller.go

```diff
@@ -376,6 +386,12 @@ func (ctrl *resizeController) pvNeedResize(pvc *v1.PersistentVolumeClaim, pv *v1
 	return true
 }
 
+// isFsResizeComplete returns true if pv.Spec.Capacity == pvc.Status.Capacity
+func (ctrl *resizeController) isFsResizeComplete(pvc *v1.PersistentVolumeClaim, pv *v1.PersistentVolume) bool {
+	klog.V(4).Infof("pv %q capacity = %v, pvc %s capacity = %v", pv.Name, pv.Spec.Capacity[v1.ResourceStorage], util.PVCKey(pvc), pvc.Status.Capacity[v1.ResourceStorage])
+	return pvc.Status.Capacity[v1.ResourceStorage].Equal(pv.Spec.Capacity[v1.ResourceStorage])
```

Comment:
We should never compare using equality for these things. It is bit like comparing floats, we will not get what we expect. I would perhaps use `pvc.Status.Capacity.Cmp(pv.Spec.Capacity) >= 0` .